runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -1230,6 +1230,8 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
          case TR::java_lang_Math_max_L:
          case TR::java_lang_Math_min_L:
             return !comp()->getOption(TR_DisableMaxMinOptimization);
+         case TR::java_lang_Math_multiplyHigh:
+            return comp()->target().is64Bit();
```

Comment:
I'm little unsure of this check is sufficient in order to enable inlining.

Reply:
I think I'd check `comp()->cg()->getSupportsLoweringConstLDiv()`. It looks like we currently generate `lmulh` only when we see `ldiv` by a constant and [replace it with multiply/shift/add](https://github.com/eclipse/omr/blob/1dd4e595421c68effbdab17f438012ebe5fd3288/compiler/optimizer/OMRSimplifierHandlers.cpp#L8984-L9009)

Maybe worth a comment in the code though to say that the lowered expression for `ldiv` uses `lmulh`

Reply:
On 2nd thought, I think `->target().is64bit()` is the best flag to use for this scenario as `SupportsLoweringConstLDiv()` is true iff [the target is 64 bits](https://github.com/eclipse/omr/blob/fc53a4c7976436dd2bae4b741ff5df6fac5c33f7/compiler/x/codegen/OMRCodeGenerator.cpp#L433).